### PR TITLE
DEL-8507: Add dashboard navigation page and private component.

### PR DESCRIPTION
**Pull request by nabinchaulagain**
## Description

Add dashboard navigation page and private component.

## JIRA Link

https://lftechnology.atlassian.net/browse/DEL-8507

## Summary of Changes

- [x] Add dashboard navigation page.
- [x] Add private component that can be used for private routes.

## Additional implementation details (OPTIONAL)

- [x] Added intermediate loading screen to prevent screen flickering when user's auth status is being checked.
- [x] Fix missing `$` in `cpBaseUrl` variable in buildspec file.

## Screenshots
image

### CODE REVIEW

File: src/routes/urls.js [hunk 1]
```diff
@@ -1,2 +1,12 @@
+import config from '../../config';
+
 export const HOME = '/';
 export const LOGIN = '/login';
+export const DASHBOARD = '/dashboard';
+
+// CP links.
+export const ACTIVITY_FEED = `${config.navigation.campaign_planning}/activity-feed`;
+export const MY_NOTIFICATION_SETTINGS = `${config.navigation.campaign_planning}/my-notification-settings`;
```
Comment (Krishuthapa) [inline src/routes/urls.js:9]: do we need this in our module?

File: src/constants/lang.js [hunk 2]
```diff
@@ -16,3 +16,20 @@ export const ERROR_MESSAGES = {
...
+    INVENTORY: 'Inventory',
+    CP: 'Campaign Planning',
+    PP: 'Protein Production',
+  },
+  SECONDARY_NAV_OPTIONS: {
+    NOTIFICATION_SETTINGS: 'Notification Settings',
+  },
+  DROPDOWN_OPTIONS: {
+    ACTIVITY_FEED: 'Activity Feed',
+    MY_NOTIFICATION_SETTINGS: 'My Notification Settings',
```
Comment (Krishuthapa) [inline src/constants/lang.js:33]: do we need this?

Comment (nabinchaulagain) [reply src/constants/lang.js:33]: We needed it for PP and CP, not sure if we need it for this module.

Should I remove this?

File: src/routes/urls.js (hunk 1 above)
Comment (nabinchaulagain) [reply src/routes/urls.js:9]: We needed it for PP and CP, not sure if we need it for this module.

image

Comment (nabinchaulagain) [reply src/routes/urls.js]: Removed

File: src/constants/lang.js (hunk 2 above)
Comment (nabinchaulagain) [reply src/constants/lang.js]: Removed

**Review by RoshanShrestha123 — APPROVED** [review]

**Pull request merged**